src/main/scala/mesosphere/util/state/PersistentStore.scala

```diff
@@ -0,0 +1,80 @@
+package mesosphere.util.state
+
+import scala.concurrent.Future
+
+/**
+  * An Entity that is provided and can be changed by the related store.
+  */
+trait PersistentEntity {
+
+  /**
+    * The identifier of this entity.
+    */
+  def id: String
+
+  /**
+    * Get the content bytes of this entity.
+    */
+  def bytes: Array[Byte]
```

Comment:
Maybe use `IndexedSeq[Byte]` instead of `Array[Byte]` to avoid exporting mutable methods.

http://stackoverflow.com/questions/3350421/what-is-most-efficient-way-to-do-immutable-byte-arrays-in-scala